### PR TITLE
Refactor mocks packages

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
### What does this PR do?

Moves Conjur mocks to dedicated directory. Binds mocks to struct instances instead of using global variables

### What ticket does this PR close?
Resolves -

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go [hunk 1]
```diff
@@ -13,22 +13,44 @@ import (
 */
 var CanExecuteConjurVar bool
```
Comment (rpothier) [inline pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go:14]: Is this left in for backwards compatability?

File: pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go [hunk 2]
```diff
@@ -13,22 +13,44 @@ import (
...
+	// 	and should probably just be an optional error.
+	Database   map[string]string
+}
+
+func (c ConjurMockClient) RetrieveSecrets (_ []byte, variableIDs []string) (map[string][]byte, error) {
 	conjurSecrets := make(map[string][]byte)
 
-	if !CanExecuteConjurVar {
+	if !c.CanExecute {
 		return nil, errors.New("custom error")
```
Comment (rpothier) [inline pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go:27]: I know this error message is existing, but could we change the error message to something better?

**Review by rpothier — COMMENTED** [review]
Looks great! just a couple minor comments.

File: pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go (hunk 1 above)
Comment (doodlesbykumbi) [reply pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go:14]: Thanks. Didn't spot this, removed!

File: pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go [hunk 3]
```diff
@@ -11,24 +11,45 @@ import (
...
-		conjurSecrets[fullVariableId] = MockConjurDB[variableId]
+		conjurSecrets[fullVariableId] = []byte(variableData)
 	}
 
 	return conjurSecrets, nil
 }
+
+func NewConjurMockClient() ConjurMockClient {
+	database := map[string]string{
+		"conjur_variable1": "conjur_secret1",
```
Comment (diverdane) [inline pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go:45]: I see this uses hardcoded, "canned" database entries. Should we add a TODO to make the database entries configurable for a new mock client object? (e.g. by either adding a `map[]string` argument to `NewConjurMockClient`, or create an `AddSecret` method for the object)?

File: pkg/secrets/k8s_secrets_storage/provide_conjur_secrets_test.go [hunk 4]
```diff
@@ -31,13 +31,13 @@ func TestProvideConjurSecrets(t *testing.T) {
 				// Sort actual and expected, because output order can change
 				sort.Strings(variableIDsActual)
 				sort.Strings(variableIDsExpected)
-				eq := reflect.DeepEqual(variableIDsActual, variableIDsExpected)
-				So(eq, ShouldEqual, true)
+
+				So(variableIDsActual, ShouldResemble, variableIDsExpected)
```
Comment (diverdane) [inline pkg/secrets/k8s_secrets_storage/provide_conjur_secrets_test.go:35]: T.I.L. Convey has a `ShouldResemble`.

**Review by diverdane — APPROVED** [review]
LGTM!!!
Thanks for taking care of this!

File: pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go (hunk 3 above)
Comment (rpothier) [reply pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go:45]: I added an AddSecret()  in [363](https://github.com/cyberark/secrets-provider-for-k8s/pull/363/commits/6e2c10d9ffdf11475e20f1d9940976797fc64646

**Pull request merged**